grafana.sh

```diff
@@ -279,14 +279,59 @@ set_vcap_datasource_prometheus() {
 }
 
 
+set_vcap_datasource_alertmanager() {
+    local datasource="${1}"
+
+    local label=$(jq -r '.label' <<<"${datasource}")
+    local name=$(jq -r '.name' <<<"${datasource}")
+    local user=$(jq -r '.credentials.prometheus.user | select (.!=null)' <<<"${datasource}")
+    local pass=$(jq -r '.credentials.prometheus.password | select (.!=null)' <<<"${datasource}")
+    local url=$(jq -r '.credentials.alertmanager.url' <<<"${datasource}")
+    local auth="true"
+
+    [[ -z "${user}" ]] && auth="false"
+
+    set +e
+    datasources_folder_exists="$(ls "${APP_ROOT}/datasources")"
+    datasources_folder_exists_exit_code=$?
+    if [[ ${datasources_folder_exists_exit_code} -eq 1 ]]
+    then
+        # Only create datasources directory if it is not already created in set_vcap_datasource_prometheus()
+        mkdir -p "${APP_ROOT}/datasources"
```

Comment:
We can execute this command even if the directory exists. It won't do anything if the directory already exists. Unless there's some logic I'm missing...

Reply:
When I test it locally, I get an error when the directory already exists:
Wenxins-MacBook-Pro:gcp-service-broker wenxinliu1$ mkdir dc/test
mkdir: dc/test: File exists

At the top of the script, `set -e` is applied. Jose wrote `set -euo pipefail`.

Equally I could simplify and just write:

```suggestion
        set +e
        mkdir -p "${APP_ROOT}/datasources"
        set -e
```
and allow the script to continue, even if it's erroring.

Here, I don't think it makes much of a difference, I feel like it's cleaner there's some way to handle the error, rather than just erroring and continuing.

So I think I'll keep this section as is.

Reply:
```
 $ mkdir -p /tmp/stuff
 $ echo $?
0
$ mkdir -p /tmp/stuff
$ echo $?
0
```
As you can see, the exit code is always 0. 

```
Wenxins-MacBook-Pro:gcp-service-broker wenxinliu1$ mkdir dc/test
mkdir: dc/test: File exists
```
You forgot the `-p` flag in your test :)

Reply:
Yes, I didn't see the `-p` flag/know about the change to functionality `-p` provides! Thank you, good point! I see the point you're making now about the unnecessary logic in this comment.

I now deleted the unnecessary logic, leaving only the `mkdir -p` line. I made a new commit that I've now pushed to this branch.